pandas/tests/indexes/interval/test_interval.py

```diff
@@ -906,6 +906,49 @@ def test_pickle_round_trip_closed(self, closed):
         tm.assert_index_equal(result, idx)
 
 
+def test_from_strings():
+    """Test the IntervalIndex.from_strings class method."""
+    # Create some float IntervalIndex string representations.
+    indices = ["(0.0, 0.5]", "(0.5, 1.0]"]
+
+    # Create some datetime-like string representations
+    datetime_indices = ["(2015-07-01, 2016-08-01]", "(2016-08-01, 2018-09-01]"]
+
+    index = IntervalIndex.from_strings(indices)
+    dt_index = IntervalIndex.from_strings(datetime_indices)
+
+    assert index[0].left == 0.0
+    assert index[1].right == 1.0
+
+    assert dt_index[0].left.year == 2015
+    assert dt_index[1].right.month == 9
+
+    # Create invalid interval indices (to make sure it fails correctly)
```

Comment:
error tests should be separate
do not catch errors rather use self.assertRaisesRegex

Reply:
Moved to a separate function in d646802. The error catching is now using a `pytest.raises` context.